python/taichi/math/mathimpl.py

```diff
@@ -10,44 +10,44 @@
 
 import taichi as ti
 
-vec2 = ti.types.vector(2, float)  # pylint: disable=E1101
-"""2D float vector type.
-"""
+_get_uint_ip = lambda: ti.u32 if impl.get_runtime(
+).default_ip == ti.i32 else ti.u64
 
-vec3 = ti.types.vector(3, float)  # pylint: disable=E1101
-"""3D float vector type.
-"""
 
-vec4 = ti.types.vector(4, float)  # pylint: disable=E1101
-"""4D float vector type.
-"""
+def vec2(*args):
+    """2D floating vector type.
+    """
+    return ti.types.vector(2, impl.get_runtime().default_fp)(*args)  # pylint: disable=E1101
```

Comment:
Nice solution. In this way it should also be fine to use `float`, though :-)

Reply:
Yes you're right, no need to call get_runtime here.